### PR TITLE
libckteec: fix in C_GetMechanismList and prevent segfault in mecha/slot info countings

### PR DESCRIPTION
**Pull request by etienne-lms**

### CODE REVIEW

File: libckteec/src/pkcs11_token.c [hunk 1]
```diff
@@ -281,7 +281,12 @@ CK_RV ck_token_mechanism_ids(CK_SLOT_ID slot,
 	if (!count || (*count && !mechanisms))
```
Comment (jenswi-linaro) [inline libckteec/src/pkcs11_token.c:281]: I suppose we should remove ` || (*count && !mechanisms)` also.

**Comment by jenswi-linaro** [comment]
For "￼libckteec: prevent segfault in C_GetSlotList()", there's a stray "fixup" at the end of the commit message.

File: libckteec/src/pkcs11_token.c (hunk 1 above)
Comment (etienne-lms) [reply libckteec/src/pkcs11_token.c:281]: you're right. I will remove it.

**Comment by etienne-lms** [comment]
updated (commit log not fixed yet)

**Comment by jenswi-linaro** [comment]
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by etienne-lms** [comment]
fixup commit squashed.
commit log message fixed.
tag applied.
thanks

**Pull request merged**